docs/AddingNewProjects.md

```diff
@@ -0,0 +1,44 @@
+# Adding New Projects To The Repo
+
+Sample PR of final result: https://github.com/dotnet/aspnetcore/pull/41945
+
+## Creating a new project
+1. Create a new folder that will house your `.csproj` and other project related files.
+2. In VS, Add a new `Solution Folder` in the same place as the new folder from the previous step.
+3. Create the project via the VS `Add` menu (select the folder -> right click -> Add -> follow the wizard).
+4. Add the following `ItemGroup` to your newly generated `.csproj`:
+    ```XML
+    <ItemGroup>
+        <None Remove="PublicAPI.Shipped.txt" />
+        <None Remove="PublicAPI.Unshipped.txt" />
+    </ItemGroup>
+    ```
+5. (OPTIONAL) Add the following line to the `.csproj`'s `PropertyGroup` to include your project in the SharedFx API:
+    ```XML
+    <IsAspNetCoreApp>true</IsAspNetCoreApp>
+    ```
+
+**Note:** Depending on what kind of project you are creating, VS will create different files in your project. Please make sure that it includes the following minimal subset of files:
+- `AssemblyInfo.cs`
+  - Lists various properties of your compiled assembly, such as to which other packages your `internal` properties and methods are available to.
+- `[Your project name].csproj`
+  - Lists various properties about your project such as dependencies, what files to compile, tags, description of the project, etc.
+- `PublicAPI.Shipped.txt`
+  - Lists publically visible APIs that are exported from your final compiled `.dll`.
+  - This only lists APIs that have already been shipped to customers and cannot be changed.
+- `PublicAPI.UnShipped.txt`
+  - Lists publically visible APIs that are exported from your final compiled `.dll`. If this is not configured properly, you will get build errors. VS will warn you though with green squiggly lines. If you see these squiggly lines, open the VS Quick Actions (CTRL + '.') and select the option to ad it to the public API.
+  - This only lists APIs that have NOT already been shipped to customers. So, these can still change.
+
+## Adding to the rest of the repo
+- VS Should have already registered your CS projec in the corresponding solution (`.sln`) and solution filter (`.slnf`) files. See this [Example](https://github.com/dotnet/aspnetcore/pull/41945/files#diff-cd977e0a76b37d35c04d9d819ea66ef8a35d9ef7f86a9a7c774d751e8119db4fR1713-R11118)
```

Comment:
Is there a way to link to an existing commit in the repo instead of a diff in a PR?

Reply:
Is this better: https://github.com/dotnet/aspnetcore/pull/41945/commits/d1fddbfbac25c529e153a08ca1a694f32a69482e

Reply:
```
https://github.com/dotnet/aspnetcore/pull/41945/commits/d1fddbfbac25c529e153a08ca1a694f32a69482e
```
^The actual link without Github magic